terraform/azure-ghaf-infra.tf

```diff
@@ -43,4 +64,62 @@ resource "azurerm_network_security_group" "ghafhydra_terraform_nsg" {
     destination_address_prefix = "*"
   }
 }
-
+# Create Storage Account
+resource "azurerm_storage_account" "ghafinfra_tf_storage_account" {
+  name                     = "ghafinfrastorage"
+  location                 = var.resource_group_location
+  resource_group_name      = azurerm_resource_group.ghaf_infra_tf_dev.name
+  account_tier             = "Standard"
+  account_replication_type = "LRS"
+}
+# Create Linux Virtual Machine
+resource "azurerm_linux_virtual_machine" "ghafinfra_tf" {
+  name                = "ghafinfratf"
+  location            = var.resource_group_location
+  resource_group_name = azurerm_resource_group.ghaf_infra_tf_dev.name
+  availability_set_id = azurerm_availability_set.ghaf_infra_tf_availability_set.id
+  network_interface_ids = [
+    azurerm_network_interface.ghaf_infra_tf_network_interface.id
+  ]
+  size = "Standard_B8ms"
+  os_disk {
+    name                 = "ghafinfratfdisk1"
+    caching              = "ReadWrite"
+    storage_account_type = "Premium_LRS"
+    disk_size_gb         = 512
+  }
+  source_image_reference {
+    publisher = "canonical"
+    offer     = "0001-com-ubuntu-server-jammy"
+    sku       = "22_04-lts-gen2"
+    version   = "latest"
+  }
+  admin_username                  = "karim"
+  disable_password_authentication = true
+  admin_ssh_key {
+    username   = "karim"
+    public_key = file("~/.ssh/id_rsa_nixos.pub")
```

Comment:
You are assuming local file `~/.ssh/id_rsa_nixos.pub` is available. I get a validation error from this line, since the file `~/.ssh/id_rsa_nixos.pub` is not available on my local system.

You should not assume `~/.ssh/id_rsa_nixos.pub` is available, but provide the value of the public key inline.

Reply:
I think we should configure the admin user public key here, as well as disable the password authentication like you already do. Then we can remove the 'customScript' resource completely.

Reply:
it's validated and and able to do the ssh